.github/workflows/ci.yml

```diff
@@ -19,8 +19,33 @@ jobs:
     - name: Run setup script
       run: ./setup.sh
 
-    - name: Build with Gradle
-      run: cd extractor &&  ./gradlew clean build
+    - name: Remove generated config folders
+      run: |
+        keep_dirs=(".github" "extractor" "infra" ".idea" ".ci-temp" "gradle")
```

Comment:
no,
you missing a nuance of how it will work.

CI should not be smart about how application work.
Application should just generate files as it suppose to do.
If application detects that file that it try to create already exists, it overrides its.
If application detects that file that it try to create already DOESNOT exists, it just create it.

So it means if any other folder or file present, application is not touching it.
Yes, we can have some trash in repo if we decide to change names of files, we can leak such files. But eventualy they will be found, as we have ability to generate all from scratch.

Reply:
One more idea, lets keep your code to remove folders at 
https://github.com/checkstyle/test-configs/blob/main/.github/workflows/commands.sh , BUT we will not use it in CI, it will be only for periodical usage by humans to cleanup repo from old files.